clang/lib/Sema/SemaBounds.cpp

```diff
@@ -4017,11 +4058,35 @@ namespace {
         << ObservedBounds << ObservedBounds->getSourceRange();
     }
 
+    // Returns the type of assignment E to show in the diagnostic messages:
+    // assignment (=), decrement (--) or increment (++).
+    Sema::BoundsDeclarationCheck GetBDCTypeFromExpr(Expr *E) const {
+      if (UnaryOperator *UO = dyn_cast<UnaryOperator>(E)) {
+        switch (UO->getOpcode()) {
```

Comment:
You can use `UnaryOperator::isIncrementOp(UO->getOpcode())` and `UnaryOperator::isDecrementOp(E->getOpcode())` to determine whether `E` is an increment operator or a decrement operator, respectively.

Reply:
Thanks for the suggestion. I found a more succinct shorthand for this: `UO->isDecrementOp()`.